### DISCUSSION TITLE
Question on plans : streaming, btree, "auto" buckets

### DISCUSSION THREAD

**Discussion by tolgaulas** [ArcadeData/arcadedb#1146] [Q&A]
Hello all,

1- Can you reveal any timeline for streaming change events in cluster mode?

2- Will you offer only LSM-tree or any plan to add B+ tree as well. (We have a lot of timeseries).
or
3- Would you consider for auto bucketing: creating a bucket if not present and adding it to the type (kind of server side create bucket + alter type with addition chain)

Thank you.

**Comment by lvca** [comment]
Hi @tolgaulas ,
1. What do you mean?
2. We have a new HSNW Index for vectors, but no plans to add a B+ tree.
3. Can you provide a practical use case for this? Right now you can create the bucket with `alter type Invoice +bucket Invoice2023`

**Reply by tolgaulas** [reply]
1- In http documentation there is this 
![image](https://github.com/ArcadeData/arcadedb/assets/1431106/a3b12bf1-a237-472b-8c10-51556a318259)

3- Yes, this means we need to check the bucket is there before creating anything in it. We insert/create a lot timeseries data along the day which we plan to store in daily buckets per customer for example. (i.e activity_customerA_20230101) that we query both with time and space (property). Very similar to postgres's timescaledb.

**Reply by lvca** [reply]
About (1) we didn't have many requests for such a feature. The main obstacle to achieving cluster-wide events is the replication protocol copies pages between servers. With such a feature enabled, the replication protocol should also send which records were touched on the page, so all the servers can fire events to their subscribers.

About (3) what do you use now? SQL?

**Reply by tolgaulas** [reply]
1- Fair. 
3- OrientDB but with indexes. I was optimist to have it here. Worst case we can make annual bucket creation for at least daily., once a year is not a big problem. 

Thanks.
